code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -70,33 +82,52 @@
 	..()
 	update_icon()
 
-/obj/machinery/atmospherics/components/unary/thermomachine/attackby(obj/item/I, mob/user, params)
-	if(!on)
-		if(default_deconstruction_screwdriver(user, icon_state_open, initial(icon_state), I))
-			return
-	if(default_change_direction_wrench(user, I))
-		return
-	if(default_deconstruction_crowbar(I))
-		return
-	return ..()
-
-/obj/machinery/atmospherics/components/unary/thermomachine/default_change_direction_wrench(mob/user, obj/item/I)
-	if(!..())
-		return FALSE
-	SetInitDirections()
+/obj/machinery/atmospherics/components/unary/thermomachine/proc/disconnect_machine()
 	var/obj/machinery/atmospherics/node = nodes[1]
 	if(node)
 		node.disconnect(src)
 		nodes[1] = null
 	nullifyPipenet(parents[1])
 
+/obj/machinery/atmospherics/components/unary/thermomachine/proc/reconnect_machine()
 	atmosinit()
-	node = nodes[1]
+	var/obj/machinery/atmospherics/node = nodes[1]
 	if(node)
 		node.atmosinit()
 		node.addMember(src)
 	build_network()
-	return TRUE
+
+/obj/machinery/atmospherics/components/unary/thermomachine/attackby(obj/item/I, mob/user, params)
+	if(anchored && !on)
+		if(default_deconstruction_screwdriver(user, icon_state_open, initial(icon_state), I))
```

Comment:
I'm gonna assume that `/obj/machinery/atmospherics/components/unary/thermomachine/heater/on` will have icon issues since it's icon_state is different from the normal variant

Reply:
Do the same with chem heaters!

Reply:
I've tested both heaters and freezers and confirmed that their icons are working as expected when on, off, and in all stages of deconstruction/moving.

Reply:
Never mind, I see what you mean, fixing it.